### ISSUE TITLE
Smarter TurboWarp button

### ISSUE THREAD

**Issue by Secret-chest** [ScratchAddons/ScratchAddons#5707]
The TurboWarp link addon should send the user to the first TW link in project notes, if it doesn't exist then it will do the same as now

**Comment by fractioneater** [comment]
People should take advantage of storing settings in the project so they don't have to put URL parameters in their turbowarp links.

**Comment by DNin01** [comment]
Is this any more efficient than just opening the TW link from the project description?

**Comment by BroJac5246** [comment]
> Is this any more efficient than just opening the TW link from the project description?

For the user, yes, especially if it's a long description.

**Comment by WorldLanguages** [comment]
Duplicate of...? I'm pretty sure this has been suggested already

**Comment by Samq64** [comment]
> Duplicate of...? I'm pretty sure this has been suggested already

I don't think there's an issue open, but there is this discussion: #537

**Comment by DNin01** [comment]
TurboWarp's "Store settings in project" feature should get the job done already?

**Comment by WorldLanguages** [comment]
> TurboWarp's "Store settings in project" feature should get the job done already?

Yeah, that's probably the best way to achieve this. It doesn't need any additional information in the project notes and the settings will be applied even when going to turbowarp.org with just the project ID and nothing else.

**Comment by Secret-chest** [comment]
> TurboWarp's "Store settings in project" feature should get the job done already?

Who uses it?

**Comment by DNin01** [comment]
> > TurboWarp's "Store settings in project" feature should get the job done already?
> 
> Who uses it?

[Vectoid TD 3D](https://scratch.mit.edu/projects/670754400/), and [Mini Golf](https://scratch.mit.edu/projects/711638071/), for example. Check the stage.

**Comment by BroJac5246** [comment]
> People should take advantage of storing settings in the project so they don't have to put URL parameters in their turbowarp links.

I agree that people should, but there are three important questions:

1. Do they store settings in projects? No.
2. Do they know how to? No.
3. Will people start doing it? Probably not.

So we need to find a solution that makes up for this.

**Comment by WorldLanguages** [comment]
Yeah, we can still do this
